File: clojure/quicksort.clj
(defn quicksort [xs]
  (if (empty? xs) xs
      (let [pivot (first xs)
            {lower false upper true} (group-by #(>= % pivot) (rest xs))]
        (concat (quicksort lower) (list pivot) (quicksort upper)))))

(def n (read))
(def xs (repeatedly n #(read)))
(println (clojure.string/join " " (quicksort xs)))

File: elixir/quicksort.exs
defmodule Quicksort do
  def quicksort([]), do: []
  def quicksort([pivot | rest]) do
    {lower, upper} = Enum.split_with(rest, &(&1 < pivot))
    quicksort(lower) ++ [pivot | quicksort(upper)]
  end
end

_ = IO.read(:stdio, :line)
xs = IO.read(:stdio, :line)
  |> String.trim
  |> String.split(" ")
  |> Enum.map(&String.to_integer/1)

Quicksort.quicksort(xs) |> Enum.join(" ") |> IO.puts
